Core/Inc/uart_ring.h

```diff
@@ -0,0 +1,21 @@
+#ifndef _UART_RING_H_
+#define _UART_RING_H_
+
+#include "stm32f051_uart.h"
+#include "ring_buffer.h"
+
+#define UART_RING_SIZE 20
+
+typedef struct
+{
+  STM32F051_UART_t uart;
+  RingBuffer_t ring;
+  uint8_t buff[UART_RING_SIZE];
+
+  void (*UART_RxCallback)();
```

Comment:
Minor: add `void` as a function parameter to show that the function doesn't get any input parameter